web/blog/2024-08-13-how-to-add-auth-with-lucia-to-your-react-nextjs-app.md

```diff
@@ -0,0 +1,639 @@
+
+For more detailed information on using Auth hooks with Wasp, visit the [Wasp documentation](https://wasp-lang.dev/docs/auth/auth-hooks).
+
+### Bonus Section: Adding Email/Password Login with Wasp and Customizing Auth
+
+Now let’s imagine we want to add email and password authentication — with all the usual features we’d expect that would follow this login method (e.g. reset password, email verification, etc.). 
+
+With Wasp, all we have to do is add a few lines to your main.wasp file, so, simply updating your Wasp configuration to include email/password authentication makes it work straight out of the box!
+
+![https://wasp-lang.dev/img/auth-ui/auth-demo-compiler.gif](https://wasp-lang.dev/img/auth-ui/auth-demo-compiler.gif)
+
+Wasp will handle the rest, also updating UI components and ensuring a smooth and secure authentication flow.
+
+```tsx
+//main.wasp
+app myApp {
+  wasp: {
+    version: "^0.14.0"
+  },
+  title: "My App",
+  auth: {
+    // 1. Specify the User entity
+    userEntity: User,
+    methods: {
+      // 2. Enable Github Auth
+      gitHub: {},
+      email: {
+        // 3. Specify the email from field
+        fromField: {
+          name: "My App Postman",
+          email: "hello@itsme.com"
+        },
+        // 4. Specify the email verification and password reset options
+        emailVerification: {
+          clientRoute: EmailVerificationRoute, //this route/page should be created
+        },
+        passwordReset: {
+          clientRoute: PasswordResetRoute, //this route/page should be created
+        },
+        // Add an emailSender -- Dummy just logs to console for dev purposes
+        // but there are a ton of supported providers :D
+        emailSender: {
+				  provider: Dummy,
+				},
+      },
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+```
+
+Implementing this in Next.js with Lucia would take a lot more work, involving a bunch of different stuff from actually sending the emails, to generating the verification tokens and more. They reference this [here](https://lucia-auth.com/guides/email-and-password/email-verification-links), but again, Wasp’s Auth makes the whole process way easier, handling a bunch of the complexity for us while also giving a bunch of other UI components, ready to use, to ease the UI details (e.g. `VerifyEmailForm`, `ForgotPasswordForm` and, `ResetPasswordForm`).
+
+The whole point here is the difference in time and developer experience in order to implement the same scenarios. For the Next.js project with Lucia, you will spend at least a few hours implementing everything if you’re going all by yourself. That same experience translates to no more than 1 hour with Wasp. What to do with the rest of the time? **Implement the important stuff your particular business requires!**
+
+---
+
+## Looking for more?
```

Comment:
I remember @matijaSos saying that we can remove the more generic "follow us"/"star us" CTAs from our own blog, but leave them for Dev.to

Reply:
I'd keep the cta, but we need to agree on whether we want to ask people to subscribe to our newsletter, or to star us on GitHub.

Reply:
ok. Just make sure to change the CTA format, i.e. remove the Dev.to specific `{ % cta ... % }` syntax.

And between asking for newsletter signups or stars, I'm not really sure. Or goal has always been stars, so stick with stars? What do you think @mmaksimovic @matijaSos?

Reply:
![Screenshot 2024-08-13 at 16 42 15](https://github.com/user-attachments/assets/61e4014e-b346-42dc-975e-51a0635cb37c)
Emotional blackmailing I did 😄

Reply:
Nice. I like it!